### PR TITLE
2.14.0

### PR DESCRIPTION
**Pull request by MegaPiggy**
- Fixed issue where menus weren't able to be used with controller/keyboard.
- Added the ability to put negative numbers in the config menu's number fields.
  - Deprecated `OnInputPopupValidateChar` and added fixed version as `OnValidateChar`. Don't think anyone had used this event yet, but just in case.
- Improved look of some config menu items.
- Fixed "MOD OPTIONS" tab closing when clicked on.
- Made "MODS" selected by default, instead of "OWML".
- Fixed text/number entry popup forcing you back to the GAMEPLAY tab.
- Added `SetValue()` and `SetText()` to IOWMLTextEntryElement.
- Made `GameObjectHelper.CreateAndAdd` use the type name if no object name is provided.

### CODE REVIEW

File: src/OWML.ModHelper.Menus/CustomInputs/OWMLTextEntryElement.cs [hunk 1]
```diff
@@ -23,9 +23,14 @@ public string GetInputText()
 			return _popup.GetInputText();
 		}
 
-		public void SetCurrentValue(string text)
```
Comment (Bwc9876) [inline src/OWML.ModHelper.Menus/CustomInputs/OWMLTextEntryElement.cs:26]: Do we know if any mods depend on this method? Would this break them?

File: src/OWML.Common/Interfaces/Menus/IOWMLPopupInputMenu.cs [hunk 2]
```diff
@@ -4,7 +4,9 @@ namespace OWML.Common.Interfaces.Menus
 {
 	public interface IOWMLPopupInputMenu
 	{
-		public event PopupInputMenu.InputPopupValidateCharEvent OnInputPopupValidateChar;
+		public delegate bool InputPopupValidateCharEvent(string input, int charIndex, char addedChar);
```
Comment (Bwc9876) [inline src/OWML.Common/Interfaces/Menus/IOWMLPopupInputMenu.cs:7]: Would this break mods that use this expecting only a `char` to be passed?

File: src/OWML.ModHelper.Menus/CustomInputs/OWMLTextEntryElement.cs (hunk 1 above)
Comment (misternebula) [reply src/OWML.ModHelper.Menus/CustomInputs/OWMLTextEntryElement.cs:26]: Dont think so, but changed back to be save.

File: src/OWML.Common/Interfaces/Menus/IOWMLPopupInputMenu.cs (hunk 2 above)
Comment (MegaPiggy) [reply src/OWML.Common/Interfaces/Menus/IOWMLPopupInputMenu.cs:7]: yes but considering no mod has used it the event yet I think it is fine

**Review by misternebula — APPROVED** [review]

**Pull request merged**